core/lib/mno_enterprise/concerns/models/ability.rb

```diff
@@ -116,6 +49,14 @@ def initialize(user)
     #
     # See the wiki for details:
     # https://github.com/CanCanCommunity/cancancan/wiki/Defining-Abilities
+
+    if user.admin_role
```

Comment:
Why such a big change? The previous idea was that:
* everyone is a regular user and can access their company
* admin have some extra abilities (to be able to connect App). But the Admin API is not really using the ability.

I understand we want to leverage the Ability to be able to do proper authorization in the admin controllers to handle the support agent but this introduces an issue as a support staff won't be able to access their company view.
Not a problem for Marlin but a problem for the core product as it'll be useful for support agents to have a demo company for training purpose.

Also, since you refactored the admin controllers to do authorization checks it'd have been good to properly implement the "staff" permissions rather than having them able to "manage all"

Reply:
Another big issue is that an admin will bypass all permissions in the customer view without having to impersonate.

Reply:
Most of this makes sense to me -- although I think adding the correct staff abilities would be too much for an already massive PR.

Reply:
As for the `can :manage, :all`, this was implicit behavior anyway for the admin endpoints.

Reply:
Second comment definitely makes sense.